src/librbd/ImageWatcher.cc

```diff
@@ -121,8 +131,19 @@ int ImageWatcher::unregister_watch() {
 
   cancel_async_requests(-ESHUTDOWN);
 
-  RWLock::WLocker l(m_watch_lock);
-  return m_image_ctx.md_ctx.unwatch2(m_handle);
+  int r = 0;
+  {
+    RWLock::WLocker l(m_watch_lock);
+    assert(m_watch_state != WATCH_STATE_UNREGISTERED);
+    if (m_watch_state == WATCH_STATE_REGISTERED) {
```

Comment:
OK, I'll look into it more.  I am pretty sure I was originally getting a seg fault within librados when I attempted to unwatch a handle that had the error handler invoked.